README.md

```diff
@@ -1,2 +1,315 @@
+```
+
+Run:
+```bash
+docker run -d -p 8005:8005 inception
+```
+
+Run from hosted image:
+```bash
+docker run -d -p 8005:8005 freelawproject/inception:v2
+```
+
+For development run it with docker compose:
+```bash
+docker compose -f docker-compose.dev.yml up
+```
+
+To handle more concurrent tasks, increase the number of workers:
+```bash
+docker run -d -p 8005:8005 -e EMBEDDING_WORKERS=4 freelawproject/inception:v2
+```
+
+Check that the service is running:
+```bash
+curl http://localhost:8005
+# Should return: "Heartbeat detected."
+```
+
+## Running tests
+```bash
+# Run all the tests
+docker exec -it inception-embedding-service pytest tests -v
+```
+```bash
+  # Run tests from a marker
+  docker exec -it inception-embedding-service pytest -m embedding_generation -v
+```
+See all available markers in [pytest.ini](pytest.ini)
+
+## API Endpoints
+
+### Query Embeddings
+Generate embeddings for search queries (CPU-optimized):
+```bash
+curl 'http://localhost:8005/api/v1/embed/query' \
+  -X 'POST' \
+  -H 'Content-Type: application/json' \
+  -d '{"text": "What are the requirements for copyright infringement?"}'
+```
+
+### Document Embeddings
+Generate embeddings for court opinions or legal documents (GPU-accelerated when available):
+```bash
+curl 'http://localhost:8005/api/v1/embed/text' \
+  -X 'POST' \
+  -H 'Content-Type: text/plain' \
+  -d 'The court finds that the defendant...'
+```
+
+### Batch Processing
+Process multiple documents in one request:
+```bash
+curl 'http://localhost:8005/api/v1/embed/batch' \
+  -X 'POST' \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "documents": [
+      {"id": 1, "text": "First court opinion..."},
+      {"id": 2, "text": "Second court opinion..."}
+    ]
+  }'
+```
+
+You can interact with the service using any HTTP client. Here's a Python example using the `requests` library:
+
+```python
+import requests
+
+# Initialize base URL
+base_url = "http://localhost:8005"
+
+# Get embedding for a query.
+response = requests.post(
+    f"{base_url}/api/v1/embed/query",
+    json={"text": "What is copyright infringement?"},
+    timeout=1,
+)
+query_embedding = response.json()["embedding"]
+
+# Get embeddings for a single document.
+response = requests.post(
+    f"{base_url}/api/v1/embed/text",
+    data="The court finds that...",
+    timeout=10,
+)
+doc_embeddings = response.json()["embeddings"]
+
+# Get embeddings for a batch of documents.
+response = requests.post(
+    f"{base_url}/api/v1/embed/batch",
+    json={
+            "documents": [
+                {"id": 1, "text": "First test document"},
+                {"id": 2, "text": "Second test document"},
+            ]
+        },
+    timeout=20,
+)
+batch_embeddings = response.json()["embeddings"]
+```
+
+
+## Configuration
+
+The service can be configured through environment variables or a `.env` file. Copy `.env.example` to `.env` to get started:
+```bash
+cp .env.example .env
+```
```

Comment:
Could this section be moved higher in the README?

I recently cloned the repository and began following the instructions. I executed the `docker compose up` command, but encountered an error because I hadn't yet created the `.env` file from the `.env.example` template.

Placing the `.env` file creation instructions earlier in the README would likely prevent this issue for other users.